stdlib/public/core/IntegerParsing.swift

```diff
@@ -146,31 +45,44 @@ extension FixedWidthInteger {
   ///     `radix`.
   ///   - radix: The radix, or base, to use for converting `text` to an integer
   ///     value. `radix` must be in the range `2...36`. The default is 10.
-  @inlinable // @specializable
-  @_semantics("optimize.sil.specialize.generic.partial.never")
+  @inlinable @inline(__always)
   public init?<S: StringProtocol>(_ text: S, radix: Int = 10) {
-    _precondition(2...36 ~= radix, "Radix not in range 2...36")
-
-    if let str = text as? String, str._guts.isFastUTF8 {
-      guard let ret = str._guts.withFastUTF8 ({ utf8 -> Self? in
-        var iter = utf8.makeIterator()
-        return _parseASCII(codeUnits: &iter, radix: Self(radix))
-      }) else {
-        return nil
-      }
-      self = ret
-      return
+    precondition(2...36 ~= radix)
+    // Check we can load the UTF-8 in UInt64 chunks.
+    let wholeGuts = text._wholeGuts
+    let canChunkLoad: Bool
+    if S.self == String.self {
+      canChunkLoad = wholeGuts.isSmall || wholeGuts.count >= 9 // +1 for sign.
+    } else {
+      canChunkLoad = (wholeGuts.isSmall ? 16 : wholeGuts.count)
+        &- text.startIndex._encodedOffset >= 9
+    }
+    if _fastPath(canChunkLoad), _fastPath(wholeGuts.isFastUTF8),
+        Self.bitWidth <= 64 {
+      let r64_ = wholeGuts.withFastUTF8({ utf8 -> UInt64? in
+        let utf8 = (S.self == String.self) ? utf8 :
+          UnsafeBufferPointer(rebasing: utf8[text._offsetRange])
+        let isSigned = Self.isSigned
+        let max = UInt64(Self.max)
+        switch radix {
```

Comment:
Side note for anyone following along; boy would it would be swell if we could precisely express the semantics that this switch should always be inlined when it's going to get eliminated, and never inlined when it won't.